src/homework9/linkedlist/LinkedList.java

```diff
@@ -0,0 +1,106 @@
+package homework9.linkedlist;
+
+public class LinkedList<T> implements List<T> {
+    private Node<T> node;
+    private int size;
+    private Node<T> prev;
+    private Node<T> first;
+    private Node<T> last;
+
+    public LinkedList() {
+        this.size = 0;
+    }
+
+    @Override
+    public void add(T value) {
+        if (size == 0) {
+            node = first = last = new Node(null, value, null);
+        } else if (size == 1) {
+            prev = node;
+            node = new Node(prev, value, null);
+            first.setNext(node);
+            last = node;
+        } else {
+            prev = node;
+            node = new Node(prev, value, null);
+            prev.setNext(node);
+            last = node;
+        }
+        size++;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        if (index > -1 && index <= size) {
+            if (index == size) {
+                this.add(value);
+            } else {
+                Node<T> necessaryNode = getNodeByIndex(index);
+                prev = necessaryNode.getPrev();
+                node = new Node<>(prev, value, necessaryNode);
+                necessaryNode.setPrev(node);
+                Node<T> prevNode = getNodeByIndex(index - 1);
```

Comment:
Is it true: prevNode == prev?
If yes, you can write: prev.setNext(node);

Reply:
I fix it.